code/_onclick/click.dm

```diff
@@ -146,16 +146,20 @@
 
 /mob/proc/click_adjacent(atom/A, obj/item/W, mods)
 	if(W)
-		if(W.attack_speed && !src.contains(A)) //Not being worn or carried in the user's inventory somewhere, including internal storages.
-			next_move += W.attack_speed
-
-		if(!A.attackby(W, src, mods) && A && !QDELETED(A))
+		var/attackby_result = A.attackby(W, src, mods)
+		var/afterattack_result
+		if(!(attackby_result & ATTACKBY_HINT_NO_AFTERATTACK) && A && !QDELETED(A))
```

Comment:
- If you're going to check if A is truthy and not deleted, you should do this first. If its not truthy it'd just be a runtime.
- Should be unnecessary to see if A is truthy and also test !QDELETED(A)